qemu/tests/check_reports_end_offset.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import re
+import json
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    'qemu-img check' should report the end offset of the image.
+    1. Create a test qcow2 image.
+    2. Check there is image end offset and the value with both "humam"
+        and "json" output.
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def _check_result(key, offset, output):
+        """Check the keywords and the value from the output."""
+        if key not in output or int(offset) != int(output[key]):
+            test.fail("The keyword/value is no correct. Check please.")
+
+    report = params["images"]
+    root_dir = data_dir.get_data_dir()
+    report = QemuImg(params.object_params(report), root_dir, report)
+    offset = params["image_end_offset"]
+    human_key = params["human_key"]
+    json_key = params["json_key"]
+
+    logging.info("Create the test image file.")
+    report.create(report.params)
+
+    """'qemu-img check' the image and check the output info."""
```

Comment:
Please comment here with `#`.
`# 'qemu-img check' the image and check the output info.`

Reply:
Updated. Check again, please.